test/http/loopback_setters_test.go

```diff
@@ -0,0 +1,466 @@
+package http
+
+import (
+	"fmt"
+	"github.com/cloudevents/sdk-go"
+	"testing"
+	"time"
+)
+
+func TestClientLoopback_setters_binary_json(t *testing.T) {
+	now := time.Now()
+
+	versions := []string{cloudevents.VersionV01, cloudevents.VersionV02, cloudevents.VersionV03}
+
+	testCases := map[string]struct {
+		event  func(string) *cloudevents.Event
+		resp   func(string) *cloudevents.Event
+		want   map[string]*cloudevents.Event
+		asSent map[string]*TapValidation
+		asRecv map[string]*TapValidation
+	}{
+		"Loopback": {
+			event: func(version string) *cloudevents.Event {
+				event := cloudevents.NewEvent(version)
```

Comment:
Check this out-->

I am creating events of all versions in a test loop.

Reply:
Nice!